events_protocol/core/utils/sync.py

```diff
@@ -0,0 +1,62 @@
+import asyncio
+from functools import wraps
+
+
+def get_current_loop():
+    return asyncio.get_event_loop()
+
+
+def make_async(fn):
+    """
+    turns a sync function to async function using threads
+    """
+    from concurrent.futures import ThreadPoolExecutor
+    import asyncio
+
+    pool = ThreadPoolExecutor()
+
+    @wraps(fn)
+    def wrapper(*args, **kwargs):
+        future = pool.submit(fn, *args, **kwargs)
+        return asyncio.wrap_future(future)  # make it awaitable
+
+    return wrapper
+
+
+def run_sync(res):
+    """Run an async method in sync way
+    CAUTION: Use only when in sync context
+    """
+    if asyncio.iscoroutine(res):
+        loop = get_current_loop()
+        return loop.run_until_complete(res)
+    return res
+
+
+def make_sync(fn):
+    """
+    turn an async function to sync function
+    """
+
+    @wraps(fn)
+    def wrapper(*args, **kwargs):
+        res = fn(*args, **kwargs)
+        return run_sync(res)
+
+    return wrapper
+
+
+class SyncMixin(object):
```

Comment:
Não precisa herdar de `object`, estamos usando Python 3
```suggestion
class SyncMixin:
```

Reply:
Classe não utilizada, removida